tests/Integration/Framework/Test_ProxyDB.py

```diff
@@ -0,0 +1,543 @@
+                                    ('/C=DN/O=DIRAC/CN=no_user', 'group', 0,
+                                     'User no exist, proxy not in DB tables'),
+                                    ('/C=DN/O=DIRAC/CN=user_1', 'group', 0,
+                                     'Group not valid, proxy not in DB tables'),
+                                    ('/C=DN/O=DIRAC/CN=user_1', 'group_1', 0,
+                                     'No proxy provider for user, proxy not in DB tables')]:
+      result = db.getProxy(dn, group, reqtime)
+      self.assertFalse(result['OK'], 'Must be fail.')
+      gLogger.info('== > %s:\nMsg: %s' % (log, result['Message']))
+    # In the last case method found proxy and must to delete it as not valid
+    cmd = 'SELECT COUNT( * ) FROM ProxyDB_Proxies WHERE UserName="user_1"'
+    self.assertTrue(bool(db._query(cmd)['Value'][0][0] == 0), "GetProxy method didn't delete proxy.")
+
+    gLogger.info('* Check that DB is clean..')
+    result = db.getProxiesContent({'UserName': ['user_ca', 'user_1', 'user_2', 'user_3']}, {})
+    self.assertTrue(result['OK'], '\n%s' % result.get('Message') or 'Error message is absent.')
+    self.assertTrue(bool(int(result['Value']['TotalRecords']) == 0), 'In DB present proxies.')
+
+    gLogger.info('* Generate proxy on the fly..')
+    result = db.getProxy('/C=DN/O=DIRACCA/OU=None/CN=user_ca/emailAddress=user_ca@diracgrid.org',
+                         'group_1', 1800)
+    self.assertTrue(result['OK'], '\n%s' % result.get('Message') or 'Error message is absent.')
+
+    gLogger.info('* Check that ProxyDB_CleanProxy contain generated proxy..')
+    result = db.getProxiesContent({'UserName': 'user_ca'}, {})
+    self.assertTrue(result['OK'], '\n%s' % result.get('Message') or 'Error message is absent.')
+    self.assertTrue(bool(int(result['Value']['TotalRecords']) == 1), 'Generated proxy must be one.')
+    for table, count in [('ProxyDB_Proxies', 0), ('ProxyDB_CleanProxies', 1)]:
+      cmd = 'SELECT COUNT( * ) FROM %s WHERE UserName="user_ca"' % table
+      self.assertTrue(bool(db._query(cmd)['Value'][0][0] == count))
+
+    gLogger.info('* Check that DB is clean..')
+    result = db.deleteProxy('/C=DN/O=DIRACCA/OU=None/CN=user_ca/emailAddress=user_ca@diracgrid.org',
+                            proxyProvider='DIRAC_CA')
+    self.assertTrue(result['OK'], '\n%s' % result.get('Message') or 'Error message is absent.')
+    result = db.getProxiesContent({'UserName': ['user_ca', 'user_1', 'user_2', 'user_3']}, {})
+    self.assertTrue(result['OK'], '\n%s' % result.get('Message') or 'Error message is absent.')
+    self.assertTrue(bool(int(result['Value']['TotalRecords']) == 0), 'In DB present proxies.')
+
+    gLogger.info('* Upload proxy..')
+    for user, dn, group, vo, time, res, log in [("user_1", '/C=DN/O=DIRAC/CN=user_1', "group_1", False, 12,
+                                                 False, 'With group extansion'),
```

Comment:
`extension` not `extansion` (here and later too)

Reply:
done